.github/workflows/ci.yml

```diff
@@ -0,0 +1,68 @@
+name: Continuous Integration
+
+on:
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+  schedule:
+    - cron: 1 11 * * *
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  unit_tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+      - name: Execute the unit tests
+        shell: bash
+        run: |
+          bazelisk test //...
+
+  remote_header_integration_test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+      - name: Configure bazelrc
+        uses: ./
+        with:
+          remote_header: "foo-bar"
+      - name: Confirm bazelrc was updated
+        shell: bash
+        run: |
+          [[ -e .bazelrc ]] || (echo >&2 ".bazelrc was not found." && exit 1)
+          grep -q -e "--remote_header=foo-bar" .bazelrc || \
+            (echo >&2 "did not find expected setting" && exit 1)
+
+  buildbuddy_api_key_integration_test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+      - name: Configure bazelrc
+        uses: ./
+        with:
+          buildbuddy_api_key: "bb-api-key"
+          bazelrc_path: .bazelrc.ci
+      - name: Confirm bazelrc was updated
+        shell: bash
+        run: |
+          [[ -e .bazelrc.ci ]] || (echo >&2 ".bazelrc.ci was not found." && exit 1)
+          grep -q -e "--remote_header=x-buildbuddy-api-key=bb-api-key" .bazelrc.ci || \
+            (echo >&2 "did not find expected setting" && exit 1)
+
+  all_ci_tests:
```

Comment:
Nice! I think the other repos could also benefit from something like this. So far we've been keeping the list of required steps up-to-date in the branch protection rules. But, collecting the status in a single job is much nicer.

Reply:
Yes! I also was thinking of this recently (as adding / removing required jobs for a bigger CI matrix is a pain in the branch protection rule UI).